File: kelaslain/Kasir.cpp
#include <iostream>
using namespace std;

int main(){
    int kode, harga, total, bayar, kembalian;
        cout << "               RUMAH MAKAN               " << endl;
        cout << "            NASI BALAP PUYUNG            " << endl;
        cout << "        Aplikasi kasir rumah makan       " << endl;
        cout << "============================================" << endl;
        cout << "" << endl;
        cout << "Menu Makanan        Harga" << endl;
        cout << "1. AYAM GORENG        Rp. 10.000" << endl;
        cout << "2. NASI GORENG        Rp.  8.000" << endl;
        cout << "3. MIE PANGSIT        Rp. 10.000" << endl;
        cout <<"MASUKKAN PILIHAN ANDA 1/2/3 : ";
        cin >> kode;

        int jml_ayam, jml_nasi, jml_pangsit;
        char tambah;
        
        //opsi 1
        if(kode == 1){
            cout << "Jumlah Ayam Goreng : ";
            cin >> jml_ayam;
            harga = 10000;
            total = harga * jml_ayam;
            cout << "Masih ada yang lain ? y/n : ";
            cin >> tambah;
            if(tambah == 'y' || tambah == 'Y'){
                cout << "Masukkan Pilihan 2/3 : ";
                cin >> kode;
                if(kode == 2){
                    cout << "Jumlah Nasi Goreng : ";
                    cin >> jml_nasi;
                    harga = 8000 * jml_nasi;
                    total += harga;
                    cout << "Ingin Memesan Pangsit Juga? y/n : ";
                    cin >> tambah;
                    if(tambah == 'y' || tambah == 'Y'){
                        cout << "Jumlah Mie Pangsit : ";
                        cin >> jml_pangsit;
                        harga = 10000 * jml_pangsit;
                        total += harga;
                    }
                }
                else if(kode == 3){
                    cout << "Jumlah Mie Pangsit : ";
                    cin >> jml_pangsit;
                    harga = 10000 * jml_pangsit;
                    total += harga;
                    cout << "Ingin Memesan Nasi Goreng Juga? y/n : ";
                    cin >> tambah;
                    if(tambah == 'y' || tambah == 'Y'){
                        cout << "Jumlah Nasi Goreng : ";
                        cin >> jml_nasi;
                        harga = 10000 * jml_nasi;
                        total += harga;
                    }
                }
            }
            
    }

    //opsi 2
    else if(kode == 2){
        cout << "Jumlah Nasi Goreng : ";
        cin >> jml_nasi;
            harga = 8000;
            total = harga * jml_nasi;
            cout << "Masih ada yang lain ? y/n : ";
            cin >> tambah;
            if(tambah == 'y' || tambah == 'Y'){
                cout << "Masukkan Pilihan 1/3 : ";
                cin >> kode;
                if(kode == 1){
                    cout << "Jumlah Ayam Goreng : ";
                    cin >> jml_ayam;
                    harga = 8000 * jml_ayam;
                    total += harga;
                    cout << "Ingin Memesan Pangsit Juga? y/n : ";
                    cin >> tambah;
                    if(tambah == 'y' || tambah == 'Y'){
                        cout << "Jumlah Mie Pangsit : ";
                        cin >> jml_pangsit;
                        harga = 10000 * jml_pangsit;
                        total += harga;
                    }
                }
                else if(kode == 3){
                    cout << "Jumlah Mie Pangsit : ";
                    cin >> jml_pangsit;
                    harga = 10000 * jml_pangsit;
                    total += harga;
                    cout << "Ingin Memesan Ayam Goreng Juga? y/n : ";
                    cin >> tambah;
                    if(tambah == 'y' || tambah == 'Y'){
                        cout << "Jumlah Ayam Goreng : ";
                        cin >> jml_ayam;
                        harga = 10000 * jml_ayam;
                        total += harga;
                    }
                }
            }
    }

    //opsi 3
    else if(kode == 3){
            cout << "Jumlah Mie Pangsit : ";
            cin >> jml_pangsit;
            harga = 10000;
            total = harga * jml_pangsit;
            cout << "Masih ada yang lain ? y/n : ";
            cin >> tambah;
            if(tambah == 'y' || tambah == 'Y'){
                cout << "Masukkan Pilihan 1/2 : ";
                cin >> kode;
                if(kode == 2){
                    cout << "Jumlah Nasi Goreng : ";
                    cin >> jml_nasi;
                    harga = 8000 * jml_nasi;
                    total += harga;
                    cout << "Ingin Memesan Ayam Goreng Juga? y/n : ";
                    cin >> tambah;
                    if(tambah == 'y' || tambah == 'Y'){
                        cout << "Jumlah Ayam Goreng : ";
                        cin >> jml_ayam;
                        harga = 10000 * jml_ayam;
                        total += harga;
                    }
                }
                else if(kode == 1){
                    cout << "Jumlah Ayam Goreng : ";
                    cin >> jml_ayam;
                    harga = 10000 * jml_ayam;
                    total += harga;
                    cout << "Ingin Memesan Nasi Goreng Juga? y/n : ";
                    cin >> tambah;
                    if(tambah == 'y' || tambah == 'Y'){
                        cout << "Jumlah Nasi Goreng : ";
                        cin >> jml_nasi;
                        harga = 10000 * jml_nasi;
                        total += harga;
                    }
                }
            }
            
    }

    //Akhir
    cout << endl << endl;
    cout << "  =====Total Keseluruhan Belanjaan =====" << endl;
    cout << "\tTotal Harga Pesanan : " << total << endl;
    cout << "\tBayar               : ";
    cin >> bayar;
    kembalian = bayar - total;
    cout << "\tKembalian           : " << kembalian << endl;
    cout << "\tTerima Kasih..." << endl;
}

File: kelaslain/kasir.cs
using System;

namespace SandBox
{
    class Program
    {
        static void Main(string[] args)
        {
            //deklarasi variabel
            int jml_ayam = 0;
            int jml_nasi = 0;
            int jml_mie = 0;
            int harga = 0;
            int total = 0;
            int bayar = 0;
            int kembalian = 0;
            char tambah;

            Console.WriteLine("               RUMAH MAKAN               ");
            Console.WriteLine("            NASI BALAP PUYUNG            ");
            Console.WriteLine("        Aplikasi kasir rumah makan       ");
            Console.WriteLine("============================================");
            Console.WriteLine();

            Console.WriteLine("Menu Makanan        Harga");
            Console.WriteLine("1. AYAM GORENG        Rp. 10.000");
            Console.WriteLine("2. NASI GORENG        Rp.  8.000");
            Console.WriteLine("3. MIE PANGSIT        Rp. 10.000");
            Console.Write("MASUKKAN PILIHAN ANDA 1/2/3 : ");
            int kode = int.Parse(Console.ReadLine());

            //opsi 1
            if(kode == 1){
                Console.Write("Jumlah Ayam Goreng : ");
                jml_ayam = int.Parse(Console.ReadLine());
                harga = 10000;
                total = harga * jml_ayam;
                Console.Write("Masih Ada yang lain? Y/n : ");
                tambah = char.Parse(Console.ReadLine());
                if(tambah == 'y' || tambah == 'Y'){
                    Console.Write("Masukkan Pilihan 2/3 : ");
                    kode = int.Parse(Console.ReadLine());
                    if(kode == 2){
                        Console.Write("Jumlah Nasi Goreng : ");
                        jml_nasi = int.Parse(Console.ReadLine());
                        harga = 8000 * jml_nasi;
                        total += harga;
                        Console.Write("Ingin Memesan Pangsit Juga? Y/n : ");
                        tambah = char.Parse(Console.ReadLine());
                        if(tambah == 'y' || tambah == 'Y'){
                            Console.Write("Jumlah Pangsit : ");
                            jml_mie = int.Parse(Console.ReadLine());
                            harga = 10000 * jml_mie;
                            total += harga;
                        }
                    }
                    else if(kode == 3){
                        Console.Write("Jumlah Mie Pangsit : ");
                        jml_mie = int.Parse(Console.ReadLine());
                        harga = 10000 * jml_mie;
                        total += harga;
                        Console.Write("Ingin Nasi Goreng Juga? Y/n : ");
                        tambah = char.Parse(Console.ReadLine());
                        if(tambah == 'y' || tambah == 'Y'){
                            Console.Write("Jumlah Nasi Goreng : ");
                            jml_nasi = int.Parse(Console.ReadLine());
                            harga = 10000 * jml_nasi;
                            total += harga;
                        }
                    }
                }
            }

            //opsi 2
            else if(kode == 2){
                Console.Write("Jumlah Nasi Goreng : ");
                jml_nasi = int.Parse(Console.ReadLine());
                harga = 8000;
                total = harga * jml_nasi;
                Console.Write("Masih Ada yang lain? Y/n : ");
                tambah = char.Parse(Console.ReadLine());
                if(tambah == 'y' || tambah == 'Y'){
                    Console.Write("Masukkan Pilihan 1/3 : ");
                    kode = int.Parse(Console.ReadLine());
                    if(kode == 1){
                        Console.Write("Jumlah Ayam Goreng : ");
                        jml_nasi = int.Parse(Console.ReadLine());
                        harga = 10000 * jml_ayam;
                        total += harga;
                        Console.Write("Ingin Memesan Pangsit Juga? Y/n : ");
                        tambah = char.Parse(Console.ReadLine());
                        if(tambah == 'y' || tambah == 'Y'){
                            Console.Write("Jumlah Pangsit : ");
                            jml_mie = int.Parse(Console.ReadLine());
                            harga = 10000 * jml_mie;
                            total += harga;
                        }
                    }
                    else if(kode == 3){
                        Console.Write("Jumlah Mie Pangsit : ");
                        jml_mie = int.Parse(Console.ReadLine());
                        harga = 10000 * jml_mie;
                        total += harga;
                        Console.Write("Ingin Ayam Goreng Juga? Y/n : ");
                        tambah = char.Parse(Console.ReadLine());
                        if(tambah == 'y' || tambah == 'Y'){
                            Console.Write("Jumlah Ayam Goreng : ");
                            jml_ayam = int.Parse(Console.ReadLine());
                            harga = 10000 * jml_ayam;
                            total += harga;
                        }
                    }
                }
            }

            //opsi 3
            else if(kode == 3){
                Console.Write("Jumlah Mie Pangsit : ");
                jml_mie = int.Parse(Console.ReadLine());
                harga = 10000;
                total = harga * jml_mie;
                Console.Write("Masih Ada yang lain? Y/n : ");
                tambah = char.Parse(Console.ReadLine());
                if(tambah == 'y' || tambah == 'Y'){
                    Console.Write("Masukkan Pilihan 1/2 : ");
                    kode = int.Parse(Console.ReadLine());
                    if(kode == 2){
                        Console.Write("Jumlah Nasi Goreng : ");
                        jml_nasi = int.Parse(Console.ReadLine());
                        harga = 8000 * jml_nasi;
                        total += harga;
                        Console.Write("Ingin Memesan Ayam Goreng Juga? Y/n : ");
                        tambah = char.Parse(Console.ReadLine());
                        if(tambah == 'y' || tambah == 'Y'){
                            Console.Write("Jumlah Ayam Goreng : ");
                            jml_ayam = int.Parse(Console.ReadLine());
                            harga = 10000 * jml_ayam;
                            total += harga;
                        }
                    }
                    else if(kode == 1){
                        Console.Write("Jumlah Ayam Goreng : ");
                        jml_ayam = int.Parse(Console.ReadLine());
                        harga = 10000 * jml_ayam;
                        total += harga;
                        Console.Write("Ingin Nasi Goreng Juga? Y/n : ");
                        tambah = char.Parse(Console.ReadLine());
                        if(tambah == 'y' || tambah == 'Y'){
                            Console.Write("Jumlah Nasi Goreng : ");
                            jml_nasi = int.Parse(Console.ReadLine());
                            harga = 10000 * jml_nasi;
                            total += harga;
                        }
                    }
                }
            }

            bool pas = false;
            do{
                Console.WriteLine("\n");
                Console.WriteLine("  =====Total Keseluruhan Belanjaan =====");
                Console.WriteLine("\tTotal Harga Pesanan : " + total);
                Console.Write("\tBayar                   : ");
                bayar = int.Parse(Console.ReadLine());
                if(bayar < total){
                    Console.WriteLine("\tMaaf Uang Anda Kurang Silahkan cukupi harga total...");
                }
                else if(bayar >= total){
                    pas = true;
                }
            }while(!pas);

            kembalian = bayar - total;

            if(bayar > total){
                Console.WriteLine($"\tKembalian           : {kembalian}");
            }

            Console.WriteLine("Terima Kasih....");
        }
    }
}
